enhancements/quota-management.md

```diff
@@ -0,0 +1,103 @@
+In the third phase, a quay admin can configure auto pruning of the images based on when they were last used to free up quota.
+
+## Motivation
+
+Users can create new organizations and repositories in their home organizations without any practical storage limitation.
+On-prem Quay customers are subject to capacity limits of their environment, specifically backend storage consumption.
+No knowledge about their storage consumption makes it hard for a service owner of a Quay registry to define SLAs and maintain a budget.
+
+### Goals
+
+* A Quay superuser can see the overall storage consumption of the entire registry i.e., an overview of all organizations depicting the aggregate storage consumption of images in an organization.
+* A Quay user can define soft quota limit on an organization that will trigger a warning when the set limit has reached.
+* A Quay user can configure alerts that will allow Quay to send notifications when set storage threshold has reached.
+* A Quay user can define hard quota limit on an organization which will not allow them to push any more images when set threshold has reached.
+
+### Non-Goals
+
+* We will not be implementing shared layer calculation i.e., if a user pushes two different versions of an image and there are some common layers between the two,
+the storage consumption is calculated independently for each of the common layers.
+
+## Design Details
+
+Initial work is to add Quota Reporting on organizations i.e., providing Quay users with their storage consumption data on a Quay organization.
+In the next steps, we will add Quota Enforcements i.e., soft and hard checks that notify or restrict Quay users from pushing images to registry.
+
+The expected design flow is depicted as below:
+![](https://user-images.githubusercontent.com/11522230/144654849-62d00b53-3be7-4009-afb6-2523f2931841.jpeg)
+NOTE: The boxes with the black border show the current flow, and the boxes with the green border show what needs to be implemented to support this feature. Design credits: @kwestpha
+
+The table RepositorySize (to be created) stores the storage consumption(in KBs) of a Quay repository within an organization. The sum of all repository sizes for an organization,
+defines the current storage size of a Quay organization.
+When an image push is initialized, the user's organization storage is validated to check if it is beyond the configured Quota limits.
+If yes, for a soft check, user is notified and, for a hard check, the push is stopped. If storage consumption is within configured Quota limits,
+the push happens as expected.
+
+Images are uploaded to the Quay registry as chunks. Each of these chunk upload triggers, the next flow.
+The size of each image chunk is stored a temporary table, TemporaryUsage (to be created) that has a foreign key reference to the earlier mentioned RepositorySize table.
```

Comment:
Do we need to track this in a separate table? BlobUpload already tracks the size of a blob upload.

Reply:
Didn't know about the BlobUpload table. That will work. Will update the design accordingly.